os_net_config/impl_nmstate.py

```diff
@@ -721,21 +721,49 @@ def cleanup_all_ifaces(self, exclude_nics=[]):
 
         logger.info("cleaning up all network configs...")
         exclude_nics.extend(['lo'])
+        exclude_nics.extend(common.get_sriov_pf_names())
+        exclude_nics.extend(common.get_dpdk_iface_names())
+        exclude_types = [OVSBridge.TYPE, OVSInterface.TYPE]
         ifaces = netinfo.show_running_config()[Interface.KEY]
-
+        logger.debug("Interface name excluded: %s", ", ".join(exclude_nics))
+        logger.debug("Interface type excluded: %s", ", ".join(exclude_types))
         for iface in ifaces:
-            if Interface.NAME in iface and \
-               iface[Interface.NAME] not in exclude_nics:
-                if iface[Interface.STATE] == InterfaceState.IGNORE:
-                    logger.info("%s: Skip cleaning", iface[Interface.NAME])
-                    continue
-                iface[Interface.STATE] = InterfaceState.ABSENT
-                state = {Interface.KEY: [iface]}
-                self.__dump_key_config(
-                    iface, msg=f"{iface[Interface.NAME]}: Cleaning up"
+            if iface.get(Interface.NAME) in exclude_nics:
+                logger.info(
+                    "%s: Interface name is excluded from cleanup",
+                    iface[Interface.NAME]
+                )
+                continue
+            if common.is_vf_by_name(iface.get(Interface.NAME)):
```

Comment:
If we were only targeting Python 3.10 and up a match statement would be good here. I would have used if/elif statements, but this works too, it's just harder to read in my opinion.